blockchain/blockvalidator.go

```diff
@@ -174,20 +174,14 @@ func (v *validator) validateActions(
 				continue
 			}
 		}
-		ctx := protocol.WithValidateActionsCtx(
-			context.Background(),
-			protocol.ValidateActionsCtx{
-				BlockHeight:  height,
-				ProducerAddr: producerAddr.String(),
-				Caller:       caller,
-			},
-		)
+		vaCtx.Caller = caller
+		aCtx := protocol.WithValidateActionsCtx(ctx, vaCtx)
```

Comment:
can use ctx = protocol.WithValidateActionsCtx(ctx, vaCtx)

Reply:
it is a multithread process, ctx will be shared. A local one is created on purpose